src/gc.h

```diff
@@ -113,27 +113,25 @@ typedef struct _mallocarray_t {
 
 // pool page metadata
 typedef struct {
-    struct {
```

Comment:
I vaguely recall we used to have exactly 8 uint16_t pointers here to ensure that `char *data` was maximally aligned. But I think that must have been from when `jl_gc_page_t` was allocated inline at that position, and is no longer applicable?